src/org/traccar/database/LdapProvider.java

```diff
@@ -16,36 +16,161 @@
+                NamingEnumeration<SearchResult> results = context.search(searchBase, searchString, searchControls);
+                if (results.hasMoreElements()) {
+                    results.nextElement();
+                    if (results.hasMoreElements()) {
+                        Log.warning("Matched multiple users for the accountName: " + accountName);
+                        return false;
+                    }
+                    return true;
+                }
+            } catch (NamingException e) {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    public InitialDirContext initContext() throws NamingException {
+        return auth(serviceUser, servicePassword);
+    }
+
+    private SearchResult lookupUser(String accountName) throws NamingException {
+        InitialDirContext context = initContext();
+
+        String searchString = new String(searchFilter).replace(":login", accountName);
+
+        SearchControls searchControls = new SearchControls();
+        String[] attributeFilter = {idAttribute, nameAttribute, mailAttribute};
+        searchControls.setReturningAttributes(attributeFilter);
+        searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+        NamingEnumeration<SearchResult> results = context.search(searchBase, searchString, searchControls);
+
+        SearchResult searchResult = null;
+        if (results.hasMoreElements()) {
+            searchResult = (SearchResult) results.nextElement();
+            if (results.hasMoreElements()) {
+                Log.warning("Matched multiple users for the accountName: " + accountName);
+                return null;
+            }
+        }
+
+        return searchResult;
+    }
+
+    public User getUser(String accountName) {
+        SearchResult ldapUser;
+        User user = new User();
+        try {
+            ldapUser = lookupUser(accountName);
+            if (ldapUser != null) {
+                Attribute attribute = ldapUser.getAttributes().get(idAttribute);
+                if (attribute != null) {
+                    user.setLogin((String) attribute.get());
+                } else {
+                    user.setLogin(accountName);
+                }
+                attribute = ldapUser.getAttributes().get(nameAttribute);
+                if (attribute != null) {
+                    user.setName((String) attribute.get());
+                } else {
+                    user.setName(accountName);
+                }
+                attribute = ldapUser.getAttributes().get(mailAttribute);
+                if (attribute != null) {
+                    user.setEmail((String) attribute.get());
+                } else {
+                    user.setEmail(accountName);
+                }
+            }
+            user.setAdmin(isAdmin(accountName));
+        } catch (NamingException e) {
+            user.setLogin(accountName);
```

Comment:
Very unlikely situation, because user was just authenticated but we should implement it.